src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+// eslint-disable-next-line import/no-cycle
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+const initialState = {
+  posts: [] as Post[],
+  hasError: false,
+  loaded: true,
+};
+
+export const setPosts = createAsyncThunk('posts/set', (userId: number) => {
+  return getUserPosts(userId);
+});
+
+export const postSLice = createSlice({
```

Comment:
The slice name `postSLice` should be renamed to `postSlice` to follow JavaScript naming conventions.